ground/src/test/java/com/google/android/ground/persistence/remote/firebase/schema/ConditionConverterTest.kt

```diff
@@ -30,6 +33,39 @@ const val TEST_OPTION_ID_C = "option-id-c-123"
 val TEST_OPTION_IDS = listOf(TEST_OPTION_ID_A, TEST_OPTION_ID_B, TEST_OPTION_ID_C)
 
 class ConditionConverterTest {
+
+  @Test
+  fun `toCondition() converts return null for empty proto`() {
+    with(ConditionConverter) { assertNull(Task.Condition.newBuilder().build().toCondition()) }
+  }
+
+  @Test
+  fun `toCondition() converts from proto`() {
+    with(ConditionConverter) {
+      val conditionProto =
+        Task.Condition.newBuilder()
```

Comment:
Suggestion:L Let's use the generated Kotlin DSL instead of the generated Java Builders for representing object literals, as they're more readable and concise. Check out `FirestoreToProtobufExtTest` for examples.

Reply:
I was trying to understand what was going on there and now my eyes are open... thank you! Done throughout, except for the one place where I use the builder pattern in `TaskConverterTest` to inject custom proto attributes (which I couldn't figure out how to do easily in the Kotlin DSL way).